### PR TITLE
Adds persistent options for the bitcask CLI

### PR DESCRIPTION
**Pull request by prologic**
Adds the same functional options to the bitcask CLI and persist options to the db store

### CODE REVIEW

**Comment by prologic** [comment]
@awnumar @jsign Any opinions on this? Don't use the CLI? This mostly useful if you want to use `bitcask` (_the command-line tool_) in shell scripts say as part of some larger task or automation/etc.

**Comment by jsign** [comment]
I didn't have the chance to use the CLI yet, but it might be helpful for scripts, yes.

**Comment by prologic** [comment]
Great! Merging then :)

File: options.go [hunk 1]
```diff
@@ -29,6 +34,43 @@ type config struct {
 	maxConcurrency  *int
 }
 
+func (c *config) MarshalJSON() ([]byte, error) {
```
Comment (jsign) [inline options.go]: This func isn't used?. Above I think you use `json.Marshal` directly.

Comment (prologic) [reply options.go]: Oh? I _could have sworn_ the docs said that it will use the `Marshaller` interface if the struct implements it? I could be wrong :)

Comment (prologic) [reply options.go]: `json.Marshal(...)` would marshal to an empty string if this doesn't work as none of the fields nor the struct are exported (https://github.com/prologic/bitcask/blob/master/options.go#L30-L35)

Comment (prologic) [reply options.go]: 🤷‍♂️🤣

**Pull request merged**